python/cudf/cudf/pylibcudf_tests/test_column_factories.py

```diff
@@ -0,0 +1,266 @@
+]
+
+DURATION_TYPES = [
+    pa.duration("s"),
+    pa.duration("ms"),
+    pa.duration("us"),
+    pa.duration("ns"),
+]
+
+DECIMAL_TYPES = [pa.decimal128(38, 2)]
+
+STRING_TYPES = [pa.string()]
+STRUCT_TYPES = [DEFAULT_STRUCT_TESTING_TYPE]
+LIST_TYPES = [pa.list_(pa.int64())]
+
+ALL_TYPES = (
+    NUMERIC_TYPES
+    + TIMESTAMP_TYPES
+    + DURATION_TYPES
+    + STRING_TYPES
+    + DECIMAL_TYPES
+    + STRUCT_TYPES
+    + LIST_TYPES
+)
+
+
+@pytest.fixture(scope="module", params=NUMERIC_TYPES, ids=repr)
+def numeric_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=DECIMAL_TYPES,
+    ids=repr,
+)
+def fixed_point_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=TIMESTAMP_TYPES,
+    ids=repr,
+)
+def timestamp_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=DURATION_TYPES,
+    ids=repr,
+)
+def duration_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        plc.MaskState.UNALLOCATED,
+        plc.MaskState.ALL_VALID,
+        plc.MaskState.ALL_NULL,
+        plc.MaskState.UNINITIALIZED,
+    ],
+    ids=["unallocated", "all_valid", "all_null", "uninitialized"],
+)
+def mask_state(request):
+    return request.param
+
+
+def test_make_empty_column_dtype(pa_type):
+    pa_col = pa.array([], type=pa_type)
+
+    # TODO: DataType.from_arrow()?
+    plc_type = plc.interop.from_arrow(pa_col).type()
+
+    if isinstance(pa_type, (pa.ListType, pa.StructType)):
+        with pytest.raises(ValueError):
+            plc.column_factories.make_empty_column(plc_type)
+        return
+
+    cudf_col = plc.column_factories.make_empty_column(plc_type)
+    assert_column_eq(cudf_col, pa_col)
+
+
+def test_make_empty_column_typeid(pa_type):
+    pa_col = pa.array([], type=pa_type)
+
+    # TODO: DataType.from_arrow()?
+    tid = plc.interop.from_arrow(pa_col).type().id()
+
+    if isinstance(pa_type, (pa.ListType, pa.StructType)):
+        with pytest.raises(ValueError):
+            plc.column_factories.make_empty_column(tid)
+        return
+
+    cudf_col = plc.column_factories.make_empty_column(tid)
+    assert_column_eq(cudf_col, pa_col)
+
+
+def test_make_numeric_column(numeric_pa_type, mask_state):
+    if mask_state == plc.types.MaskState.ALL_NULL:
+        expected = pa.array([None] * EMPTY_COL_SIZE, type=numeric_pa_type)
+    else:
+        # TODO: uninitialized not necessarily 0
+        expected = pa.array(
+            [0 if numeric_pa_type is not pa.bool_() else False]
+            * EMPTY_COL_SIZE,
+            type=numeric_pa_type,
+        )
+
+    plc_type = plc.interop.from_arrow(numeric_pa_type)
+
+    got = plc.column_factories.make_numeric_column(
+        plc_type, EMPTY_COL_SIZE, mask_state
+    )
+    assert_column_eq(got, expected)
+
+
+@pytest.mark.parametrize(
+    "non_numeric_pa_type", list(set(ALL_TYPES) - set(NUMERIC_TYPES))
```

Comment:
*issue*: Please ensure that this parameterisation has a consistent order, otherwise collection in parallel (with xdist) will fail.

```
[t for t in ALL_TYPES if t not in NUMERIC_TYPES]
```